### PR TITLE
Extract mock HTTP response body to a function

### PR DESCRIPTION
**Pull request by Glutexo**
DRY

### CODE REVIEW

**Comment by nappex** [comment]
My opinion is that this PR is unnecessary because of #41 . PR #41  is solving DRY of the body already.

**Comment by Glutexo** [comment]
#41 only dries the mock `get!/1` function. The body composition is however a slightly different thing. The `get!/1` function is only one place from where it is used. The other place are the body asserts in the tests, where the composition logic is repeated.

In

Then in the tests, the expected body can be generated by

```elixir
assert(body(url) == File.read!(path))
```

instead of

```elixir
assert("Body from: #{url}\n" == File.read!(path))
```

**Comment by Glutexo** [comment]
Merged [_elixir_](https://github.com/Glutexo/onigumo/tree/elixir) into this, so the change now becomes more apparent.

**Comment by nappex** [comment]
> #41 only dries the mock `get!/1` function. The body composition is however a slightly different thing. The `get!/1` function is only one place from where it is used. The other place are the body asserts in the tests, where the composition logic is repeated.
> 
> In the `get/1` function, the body can be generated by
> 
> ```elixir
> %HTTPoison.Response{
>   status_code: 200,
>   body: body(url)
> }
> ```
> 
> Then in the tests, the expected body can be generated by
> 
> ```elixir
> assert(body(url) == File.read!(path))
> ```
> 
> instead of
> 
> ```elixir
> assert("Body from: #{url}\n" == File.read!(path))
> ```
> 
> .

Ok I've got it.

**Comment by Glutexo** [comment]
Merged current [_elixir_](https://github.com/Glutexo/onigumo/tree/elixir) into this. Resolved conflicts.

**Review by nappex — APPROVED** [review]

File: test/onigumo_test.exs [hunk 1]
```diff
@@ -34,7 +35,11 @@ defmodule OnigumoTest do
   defp get!(url) do
     %HTTPoison.Response{
       status_code: 200,
-      body: "Body from: #{url}\n"
+      body: body(url)
     }
   end
+
+  defp body(url) do
+    "Body from: #{url}"
```
Comment (Glutexo) [inline test/onigumo_test.exs:43]: This was my mistake when merging. 😞 It should have already contain the newline.

```suggestion
    "Body from: #{url}\n"
```

**Review by Glutexo — COMMENTED** [review]
Spotted post-mortem mistake. 😞

File: test/onigumo_test.exs (hunk 1 above)
Comment (Glutexo) [reply test/onigumo_test.exs:43]: Fixed in https://github.com/Glutexo/onigumo/commit/35cce947a85f99114abad41b4314c08a61f42068.

**Comment by Glutexo** [comment]
Fixed in https://github.com/Glutexo/onigumo/commit/35cce947a85f99114abad41b4314c08a61f42068 directly pushed to master.

**Comment by Glutexo** [comment]
Fixed in https://github.com/Glutexo/onigumo/commit/35cce947a85f99114abad41b4314c08a61f42068 directly pushed to [_elixir_](https://github.com/Glutexo/onigumo/tree/elixir).

**Pull request merged**